### PR TITLE
paych: check To address is owned by wallet for inbound channels

### PR DESCRIPTION
**Pull request by dirkmc**

### CODE REVIEW

File: paychmgr/manager.go [hunk 1]
```diff
@@ -218,14 +199,76 @@ func (pm *Manager) CheckVoucherSpendable(ctx context.Context, ch address.Address
...
+
+	// Channel is not in store, so get channel from state
+	stateCi, err := pm.sa.loadStateChannelInfo(ctx, ch, DirInbound)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that channel To address is in wallet
+	to := stateCi.Control // Inbound channel so To addr is Control (this node)
+	has, err := pm.pchapi.WalletHas(ctx, to)
```
Comment (dirkmc) [inline paychmgr/manager.go:259]: This passes the integration test, so I believe it's correct to call `WalletHas()` with the channel To address, as opposed to calling it with the channel To address account's `Address` field.
@magik6k can you confirm?

Comment (magik6k) [reply paychmgr/manager.go:259]: WalletHas expects a key address (`t1..`, `t3..`), so assuming it's guaranteed to be that, it's fine

Comment (dirkmc) [reply paychmgr/manager.go:259]: [This is the address](https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/paych/paych_state.go#L18) that will be passed to WalletHas()

ie the "To" address of the channel state
https://github.com/filecoin-project/specs-actors/blob/a73a22a48b668c2f5a1cc6a31caef03bcaec5f70/actors/builtin/paych/paych_state.go#L11-L30
```
type State struct {
	// Channel owner, who has funded the actor
	From addr.Address
	// Recipient of payouts from channel
	To addr.Address

	// Amount successfully redeemed through the payment channel, paid out on `Collect()`
	ToSend abi.TokenAmount

	// Height at which the channel can be `Collected`
	SettlingAt abi.ChainEpoch
	// Height before which the channel `ToSend` cannot be collected
	MinSettleHeight abi.ChainEpoch

	// Collections of lane states for the channel, maintained in ID order.
	LaneStates cid.Cid // AMT<LaneState>
}
```

Comment (magik6k) [reply paychmgr/manager.go:259]: Looks like the actor code normalizes those to ID addresses, so we need to call StateAccountKey on those first

**Review by magik6k — APPROVED** [review]

**Pull request merged**